src/executables/ddl_proxy_client.cpp

```diff
@@ -0,0 +1,255 @@
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation; either version 2 of the License, or
+ *   (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program; if not, write to the Free Software
+ *  Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+ *
+ */
+
+
+/*
+ * ddl_proxy_client.cpp - client that executes ddl queries
+ */
+
+#include "util_support.h"
+#include "cubrid_getopt.h"
+#include "utility.h"
+#include "csql.h"
+#include "db.h"
+#include "dbi.h"
+
+static void
+utility_print (int message_num, ...)
+{
+  typedef const char *(*GET_MESSAGE) (int message_index);
+
+  DSO_HANDLE util_sa_library;
+  DSO_HANDLE symbol;
+  GET_MESSAGE get_message_fn;
+
+  utility_load_library (&util_sa_library, LIB_UTIL_SA_NAME);
+  if (util_sa_library == NULL)
+    {
+      utility_load_print_error (stderr);
+      return;
+    }
+  utility_load_symbol (util_sa_library, &symbol, UTILITY_GENERIC_MSG_FUNC_NAME);
+  if (symbol == NULL)
+    {
+      utility_load_print_error (stderr);
+      return;
+    }
+
+  get_message_fn = (GET_MESSAGE) symbol;
+
+  {
+    va_list ap;
+
+    va_start (ap, message_num);
+    vfprintf (stderr, get_message_fn (message_num), ap);
+    va_end (ap);
+  }
+}
+
+static int start_ddl_proxy_client(const char *program_name, CSQL_ARGUMENT *args)
+{
+  DB_SESSION *session = NULL;
+  int rc = NO_ERROR;
+
+  rc = db_restart_ex (program_name, args->db_name, args->user_name, args->passwd, NULL, DB_CLIENT_TYPE_ADMIN_CSQL);
+
+  if (rc != NO_ERROR)
+    {
+      return rc;
+    }
+
+  if (args->command != NULL)
+    {
+      int total_stmts, stmt_id, i, num_of_rows;
+      DB_QUERY_RESULT *result = NULL;
+
+      session = db_open_buffer ((const char *) args->command);
+      if (session == NULL)
+	{
+          rc = er_errid ();
+	  goto error;
+	}
+
+        if (db_get_errors (session) || er_errid () != NO_ERROR)
+          {
+            rc = er_errid ();
+            goto error;
+          }
+
+        total_stmts = db_statement_count (session);
+        for (i = 0; i < total_stmts; i++)
+          {
+            stmt_id = db_compile_statement (session);
+            if (stmt_id < 0)
+              {
+                rc = er_errid ();
+                db_abort_transaction ();
+                goto error;
+              }
+
+            if (stmt_id == 0)
+              goto error;
+
+            num_of_rows = db_execute_statement (session, stmt_id, &result);
+            if (num_of_rows < 0)
+              {
+                rc = er_errid ();
+                db_abort_transaction ();
```

Comment:
Just as remainder for future :
The final solution may need another implementation then abort_transaction.